src/track/serato/markers2.cpp

```diff
@@ -2,6 +2,23 @@
 
 #include <QtEndian>
 
+namespace {
+QString zeroTerminatedUtf8StringtoQString(QDataStream* stream) {
+    DEBUG_ASSERT(stream);
+
+    QByteArray data;
+    quint8 byte = '\xFF';
+    while (byte != '\x00') {
```

Comment:
did you consider to use QDataStream &QDataStream::operator>>(char *&s)

Reply:
Yes, but it doesn't work. It seems to expect a 4 byte length prefix or something like that. I always got "Read past end".